src/_P003_Pulse.ino

```diff
@@ -455,34 +459,27 @@ void doPulseStepProcessing(int pStep, taskIndex_t taskIndex, uint16_t debounceTi
     case P003_PROCESSING_STEP_0:   
     // regularily called to check if the trigger has flagged the next signal edge
     {
-      // check if flag for any task
-      if( P003_initStepsFlags )
+      if( bitRead(P003_initStepsFlags, taskIndex) )
       {
-        for ( int i_taskID = 0; i_taskID < TASKS_MAX; i_taskID++ )
-        {
-          if( bitRead(P003_initStepsFlags, i_taskID) )
-          {
-            // schedule step 1 in remaining milliseconds from debounce time
-            long delayTime = (long) (P003_debounceTime[i_taskID]-(getMicros64()-P003_triggerTimestamp[i_taskID]))/1000L;
-            if(delayTime < 0)
-            {              
-              // if debounce time was too short or we were called too late by Scheduler
-              #ifdef P003_PULSE_STATISTIC
-                P003_Step0ODcounter[i_taskID]++;  // count occurences
-                P003_StepOverdueMax[i_taskID][pStep] = max(P003_StepOverdueMax[i_taskID][pStep], -delayTime);
-              #endif // P003_PULSE_STATISTIC
-              delayTime = 0;
-            }
-            Scheduler.setPluginTaskTimer(delayTime, i_taskID, P003_PROCESSING_STEP_1);
-            
-            #ifdef P003_PULSE_STATISTIC
-              P003_triggerTimestamp[i_taskID] = getMicros64() + delayTime*1000L;
-            #endif // P003_PULSE_STATISTIC
-
-            // initialization done 
-            bitClear(P003_initStepsFlags, i_taskID);
-          }
+        // schedule step 1 in remaining milliseconds from debounce time
+        long delayTime = (long) (P003_debounceTime[taskIndex]-(getMicros64()-P003_triggerTimestamp[taskIndex]))/1000L;
```

Comment:
Not sure if this calculation is always as expected when `delayTime < 0`
You're computing using unsigned values, so the result may not be predictable.

You are only expecting a negative value, so maybe better to check like this:
```c++
const long delayTime = static_cast<long>((static_cast<int64_t>(P003_debounceTime[taskIndex])-static_cast<int64_t>(getMicros64()-P003_triggerTimestamp[taskIndex]))/1000L);
```

Reply:
I tested the simple casting to (long) of the previous version intensively and it was working. My understanding is, that addition and subtraction for unsigned and signed generate identical result. the only difference is, that for signed the MSBit is interpreted as negative.
But I took your version to be save and it is working either.
unsigned and signed